_posts/2019-06-19-python-gpus-status-update.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: post
+title: "Python and GPUs: A Status Update"
+author: Matthew Rocklin
+tags: [python,scipy]
+theme: twitter
+---
+{% include JB/setup %}
+
+*This blogpost was delivered in talk form at the recent [PASC
+2019](https://pasc19.pasc-conference.org/) conference.
+[Slides for that talk are
+here](https://docs.google.com/presentation/d/e/2PACX-1vSajAH6FzgQH4OwOJD5y-t9mjF9tTKEeljguEsfcjavp18pL4LkpABy4lW2uMykIUvP2dC-1AmhCq6l/pub?start=false&loop=false&delayms=60000).*
+
+Executive Summary
+-----------------
+
+We're improving the state of scalable GPU computing in Python.
+This post lays out the current status broadly, and describes future work.  It
+also summarizes several other more focused blogposts from recent months, and
+includes links for the interested reader.
+
+Broadly we cover briefly the following categories:
+
+-  Python libraries written in CUDA like CuPy and RAPIDS
+-  Python-CUDA compilers, specifically Numba
+-  Scaling these libraries out with Dask
+-  Network communication with UCX
+-  Packaging with Conda
+
+
+Performance of GPU accelerated Python Libraries
+-----------------------------------------------
+
+Libraries like [CuPy](https://cupy.chainer.org/) and
+[RAPIDS](https://rapids.ai/) build GPU accelerated variants of popular Python
+libraries like NumPy, Pandas, and Scikit-Learn.  In order to better understand
+the relative performance differences [Peter
+Entschev](https://github.com/pentschev) recently put together a [benchmark
+suite](https://github.com/pentschev/pybench) to help with comparisons.  He has
+produced the following images:
+
+TODO
```

Comment:
FYI @pentschev .  No rush on this, but I figured that this might be another place where your work would have some impact.

Reply:
Sure, that would probably be nice. I still have to push some more code and probably some info on how to use that too. :)

Reply:
How soon are you planning to have this published

Reply:
@pentschev I think that the goal is to push sometime out tomorrow.